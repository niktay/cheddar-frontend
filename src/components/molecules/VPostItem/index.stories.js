import { storiesOf } from "@storybook/vue";
import VPostItem from ".";

storiesOf("Molecule - VPostItem", module).add("default", () => ({
  components: { VPostItem },

  data() {
    return {
      post: {
        title: "Sample Title",
        author: {
          firstName: "John",
          lastName: "Doe"
        },
        content: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptate mollitia quaerat voluptatem 
                  optio obcaecati quos ex saepe. Minus, optio quis! Labore fugit deleniti ex id commodi ab nobis eaque 
                  adipisci?`
      }
    };
  },

  template: `<VPostItem :post="post"/>`
}));
