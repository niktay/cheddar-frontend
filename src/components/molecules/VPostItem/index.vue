<template>
  <article class="m-VPostItem">
    <VPostTitle>{{ post.title }}</VPostTitle>
    <VPostAuthor :author="post.author" />
    <VPostContent>{{ post.content }}</VPostContent>
  </article>
</template>

<script>
import VPostTitle from "@/components/atoms/VPostTitle/.";
import VPostAuthor from "@/components/atoms/VPostAuthor/.";
import VPostContent from "@/components/atoms/VPostContent/.";

export default {
  components: {
    VPostTitle,
    VPostAuthor,
    VPostContent
  },

  props: {
    post: {
      type: Object,
      required: true
    }
  }
};
</script>

<style lang="scss">
</style>
