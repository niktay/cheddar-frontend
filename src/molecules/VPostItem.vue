<template>
  <article class="m-VPostItem">
    <VPostTitle>{{ post.title }}</VPostTitle>
    <VPostAuthor :author="post.author" />
    <VPostContent>{{ post.content }}</VPostContent>
  </article>
</template>

<script>
import VPostTitle from "@/atoms/VPostTitle";
import VPostAuthor from "@/atoms/VPostAuthor";
import VPostContent from "@/atoms/VPostContent";

export default {
  components: {
    VPostTitle,
    VPostAuthor,
    VPostContent
  },

  props: {
    post: {
      type: Object,
      required: true
    }
  }
};
</script>

<style lang="scss">
</style>
