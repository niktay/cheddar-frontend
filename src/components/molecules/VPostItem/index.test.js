import { mount } from "@vue/test-utils";
import VPostItem from ".";

describe("VPostItem component", () => {
  it("should render correctly", () => {
    const comp = mount(VPostItem, {
      propsData: {
        post: {
          title: "Sample Title",
          content: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Aperiam hic esse sit atque enim fugiat 
                    provident delectus sunt quos. Suscipit nisi iusto ducimus dolorem minus quo adipisci deleniti esse 
                    obcaecati!`,
          author: {
            firstName: "John",
            lastName: "Doe"
          }
        }
      }
    });

    expect(comp.html()).toMatchSnapshot();
  });
});
