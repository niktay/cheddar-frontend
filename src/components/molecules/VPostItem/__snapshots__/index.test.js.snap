// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VPostItem component should render correctly 1`] = `
<article class="m-VPostItem">
  <h2 class="a-VPostTitle">Sample Title</h2> <span class="a-VPostAuthor">
  by John Doe
</span>
  <p class="a-VPostContent">Lorem ipsum dolor, sit amet consectetur adipisicing elit. Aperiam hic esse sit atque enim fugiat
    provident delectus sunt quos. Suscipit nisi iusto ducimus dolorem minus quo adipisci deleniti esse
    obcaecati!</p>
</article>
`;
